test/testpermissions.cpp

```diff
@@ -499,11 +496,16 @@ private slots:
         lm.rename("changeonly/sub2/filetorname2a", "changeonly/sub2/aaa2_renamed");
         lm.rename("changeonly/sub2/filetorname2z", "changeonly/sub2/zzz2_renamed");
 
+        auto expectedState = fakeFolder.currentLocalState();
+
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), expectedState);
+        QCOMPARE(fakeFolder.currentRemoteState(), expectedState);
+
```

Comment:
why do you put an extra `syncOnce()` call here ?
are you not changing what this test is doing ?

Reply:
@mgallien There are no `syncOnce()` calls in this test. I call it each time I do changes either to local or to remote state, so changes do propagate from one to another state. There are 3 calls to `syncOnce()` in this test. The first one is called after creating a folder structure via remote state `rm.mkdir(), etc.`, the second one is called after renaming some files, while the third one is called after renaming some folders. Please double-check.

Yes, you are right. I have to change what this test is doing. This is because we are now restoring all nested files/folders in case the parent was moved/renamed while not allowed. This effectively prevents deleting each nested file/folder one by one in case they were allowed to be removed, but, are not supposed to get removed when the parent is renamed.